src/gt4py/eve/concepts.py

```diff
@@ -234,3 +237,22 @@ class GenericNode(datamodels.GenericDataModel, Node, kw_only=True):  # type: ign
 def eq_nonlocated(a: Node, b: Node) -> bool:
     """Compare two nodes, ignoring their `SourceLocation` or `SourceLocationGroup`."""
     return len(utils.ddiff(a, b, exclude_types=[SourceLocation, SourceLocationGroup])) == 0
+
+
+def selective_pickler(skipped_fields: Container[str]) -> type:
```

Comment:
I would make this  function cacheable to avoid creating a new class per call.

```suggestion
@functools.cache
def selective_pickler(*skipped_fields: str) -> type:
```